Monika After Story/game/script-story-events.rpy

```diff
@@ -269,6 +269,202 @@ label monika_changename:
+
+        persistent._mas_player_confirmed_bday = True
+
+    if old_bday is not None:
+        $ old_bday_not_none = old_bday.replace(year=mas_player_bday_curr.year)
+
+    if not mas_isplayer_bday() and old_bday_not_none == mas_player_bday_curr:
+        return
+
+    if mas_isplayer_bday():
+        if old_bday_not_none == mas_player_bday_curr:
+            if mas_isMoniNormal(higher=True):
+                m 3hub "Ahaha! So today {i}is{/i} your birthday!"
+                m 1tsu "I'm glad I was prepared, ehehe..."
+                m 3eka "Hold on just one moment, [player]..."
+                show monika 1dsc
+                pause 2.0
+                jump mas_player_bday_when_confirmed
+            elif mas_isMoniDis(higher=True):
+                $ persistent._mas_player_bday_low_aff = True
+                m 2eka "Ah, so today {i}is{/i} your birthday..."
+                m "Happy Birthday, [player]."
+                m 4eka "I hope you have a good day."
+        else:
+            $ persistent._mas_player_bday_when_found = True
+            if mas_isMoniNormal(higher=True):
+                $ mas_unlockEVL("bye_player_bday", "BYE")
+                m 1wuo "Oh...{w=1}Oh!"
+                m 3sub "Today's your birthday!"
+                m 3hub "Happy Birthday, [player]!"
+                m 1rksdla "I wish I had known earlier so I could've prepared something."
+                m 1eka "But I can at least do this..."
+                call mas_player_bday_moni_sings
+                m 1hub "Ahaha! It's not much but it's something!"
+                m 3hua "I promise next year we'll do something extra special, [player]!"
+            elif mas_isMoniDis(higher=True):
+                $ persistent._mas_player_bday_low_aff = True
+                m 2eka "Oh, so today's your birthday..."
+                m "Happy Birthday, [player]."
+                m 4eka "I hope you have a good day."
+        return
+
+    # have to use the raw data here to properly compare in the rare even that the player bday and first sesh are on 2/29
+    if persistent._mas_player_bday.month == persistent.sessions.get("first_session",None).month and persistent._mas_player_bday.day == persistent.sessions.get("first_session",None).day:
+        m 1sua "Oh! Your birthday is the same date as our anniversary, [player]?"
+        m 3hub "That's amazing!"
+        m 1sua "I can't imagine a more special day than celebrating your birthday and our love on the same day..."
+        if mas_player_bday_curr == mas_o31 or mas_player_bday_curr == mas_d25 or mas_player_bday_curr == mas_monika_birthday:
+            #TODO: add more holidays here (f14)
+            if mas_player_bday_curr == mas_o31:
+                $ hol_str = "Halloween"
+            if mas_player_bday_curr == mas_d25:
+                $ hol_str = "Christmas"
+            if mas_player_bday_curr == mas_monika_birthday:
+                $ hol_str = "my birthday"
+            m "And with it also being [hol_str]..."
```

Comment:
rather than having a comparison check twice per holiday, this woudl be more efficient:
```python
if bday == o31:
    # set hol str
elif bday == d25:
    # set hol str
elif bday == 922:
    # set hol str
else:
    # set hol str to None

if hol str is not None:
    m "And with it also being..."
```